modules/hostap/src/supp_api.c

```diff
@@ -1550,6 +1550,106 @@ int supplicant_ap_config_params(const struct device *dev, struct wifi_ap_config_
 }
 #endif
 
+#ifdef CONFIG_WIFI_NM_HOSTAPD_WPS
```

Comment:
Even these functions are identical to WPA supp based WPS except for using hostapd_cli, so, makes sense to avoid duplication.

Reply:
not only the hostapd_cli, also get_hostapd_handle is different. Separate the WPS case for STA and SAP is more clean way,  as in the future, the hostapd code may be split into another file.

Reply:
Hi @krish2718  there are still coming PRs from our downstream, better add supp_api_wps.c... in the future. thanks

Reply:
Sure, I was only mentioning it for the future, we already have a task https://github.com/zephyrproject-rtos/zephyr/issues/77383 we can keep adding comments there so that we don't miss.